content/blog/which-unique-identifier-is-right-for-you.md

```diff
@@ -30,75 +29,73 @@ Let's break down UUID for better visibility.
-- Due to limited randomness, It doesn't guarantee a collision-resistant solution.
-- It has yet to be adopted much due to compatibility issues with UUID. (if systems are already using UUID).
+- Randomness is limited in the timestamp, so it is possible to generate multiple IDs within the same millisecond.
+- Due to limited randomness, it doesn't guarantee a collision-resistant solution.
+- Not widely adopted due to compatibility issues with UUID systems.
 
 ---
 
-### CUID
+### CUID (Collision-Resistant Unique Identifier)
 
-Collision-resistant Unique Identifier also known as CUID, It's secure, guarantees strong collision-resistant and provides good performance.
+CUIDs are designed for high collision resistance and performance, with a default length of 24 characters but configurable up to 32 characters.
 
 To react 50% chance of collision, you'd need to generate roughly 4,000,000,000,000,000,000 IDs. Which is quite huge.
 
-It is configurable to at max 32 characters long, the default length is 24 characters.
-
-Let's break it down to understand it better
-
 {{< figure src="/images/2024/which-unique-identifier-is-right-for-you/CUID-architecture.png" >}}
 
-It uses a combination of UNIX Time in Millisecond + Salt + Session Count + Fingerprint and Hash these values with SHA3 Hashing Algorithm prefix by random alphabet.
+It uses a combination of UNIX time in milliseconds, salt, session count, fingerprint, and hashes these values with the SHA3 hashing algorithm, prefixed by a random alphabet.
 
 #### Pros
 
-- The main advantage of CUID is Collision-Resistant, It's quite strong in ensuring uniqueness. It focuses on keeping entropy as high as possible to ensure correct randomness.
-- It's secure and non-guessable, provides offline ID generation support, URL friendly (Doesn't contain special symbols).
-- CUID is horizontally scalable, it can generate IDs on various machines with coordination.
+- Strong collision resistance, making it highly unique.
+- Secure, non-guessable, URL-friendly, and supports offline ID generation.
+- Horizontally scalable, can generate IDs across multiple machines.
 
 #### Cons
 
 - It doesn't work well if sequentiality is in focus.
-- If security and cross-host uniqueness are not priorities then CUID doesn't perform well.
-- It's a little complex due to SHA3 Hashing and Pseudorandom Number Generator (PRNG) compared to UUID.
+- Less performant if security and cross-host uniqueness are not priorities.
+- Complex due to SHA3 hashing and Pseudorandom Number Generator (PRNG).
+
 
 ---
 
 ### NanoId
 
-It is a tiny, secure, URL-friendly, unique string ID generator for JavaScript. NanoId has a similar number of random bits, hence, It has similar collision probability as UUID.
+NanoID is a tiny, secure, URL-friendly, unique string ID generator for JavaScript. With a similar number of random bits, NanoID has a collision probability comparable to UUID.
```

Comment:
"**NanoId has similar collision probability as UUID**"

Both have similar mechanisms, hence have an equal chance of collision